R/dataframe.R

```diff
@@ -200,6 +200,34 @@ add_rownames <- function(df, var = "rowname") {
   as_data_frame(cbind(rn, df))
 }
 
+#' Convert a column of tbl into rownames of data.frame.
+#'
+#' Inverted operation of add_rownames
+#'
+#' @param df Input data frame with rownames.
+#' @param var Name of variable to use
+#' @export
+#' @seealso \code{\link{add_rownames}}
+#' @examples
+#' tbl_df(mtcars)
+#'
+#' mtcars_tbl <- add_rownames(mtcars)
+#' mtcars_data.frame <- use_as_rownames(mtcars_tbl)
+use_as_rownames <- function(df, var = "rowname") {
+  stopifnot(is.data.frame(df))
+
+  if ( !var %in% colnames(df) ) {
+    stop(paste("No", var, "column in the colnames.") )
+  }
+
+  df <- as.data.frame(df)
+  rownames(df) <- df[[var]]
```

Comment:
1. if a data frame has row names, users intend not to use `use_as_rownames`. If used, just update it with the `var` column in the data frame. 
2. In this situation, `add_rownames` will add another `var` column. I suggest that return stop if there is a `var` in the input of `add_rownames()`. So far, `tbl_df()` keeps the rownames, though not shown. `add_rownames()` returns a character number as rownames. So we can keep the original rownames for the return value of `add_rownames()`.